packages/expo-modules-core/ios/Core/DynamicTypes/DynamicConvertibleType.swift

```diff
@@ -18,6 +18,9 @@ internal struct DynamicConvertibleType: AnyDynamicType {
   }
 
   func cast<ValueType>(_ value: ValueType, appContext: AppContext) throws -> Any {
+    if value is Convertible {
```

Comment:
I know that's coming from my initial changes, but could you try to remove it and see what's happening? I don't remember why I needed to add this 🤔

Reply:
I've removed it but I'm not seeing any issues now. I'll add it back it I run into something related

Reply:
It's still there 😄 I think it's no longer needed, but if it is, then this is not the best solution to solve it.